subscription/notify_lambda/notification-lambda.py

```diff
@@ -0,0 +1,41 @@
+import json
+import requests
+from logger import logger
+
+def handler(event, context):
+    logger.debug(f""Received event: {json.dumps(event, indent=2)}"
+    for record in event['Records']:
+        process_message(record)
+
+def process_message(record):
+    try:
+        message = record['Sns']['Message']
+        message_attributes = record['Sns']['MessageAttributes']
+        logger.info(f"Processing message {message}")
+        send_message(message_attributes['endpoint'], message)
+        
+    except Exception as e:
+        logger.error(f"An error occurred {e}")
+        raise e
+
+def send_message(url, message):
+    # Prepare the data to be sent
+
+    try:
+        # Send a POST request to the URL with the message data
+        headers = {'Content-Type': 'application/json'}
+        logger.debug("URL: ", url)
+        logger.debug("Message: ", message)
+        response = requests.post(url, headers=headers, json=message)
+
+        # Check if the request was successful
+        if response.status_code == 200:
+            logger.debug("Message sent successfully!")
+            logger.debug(f"Response: {response.text}")
+        else:
+            logger.error(f"Failed to send message. Status code: {response.status_code}")
+            logger.debug(f"Response: {response.text}")
```

Comment:
do you need a sys.exit(1) to ensure that the lambda marks itself as having failed? Though I guess it might depend on what kind of error occurred, since it could simply be an error on the other end of the request

Reply:
Or perhaps the fact that it sent the request regardless of return value is success as far as the lambda is concerned. Generally I'm trying to think if there's a way we could know how many lambdas are getting non 2xx responses from sending the data

Reply:
I could put the data that fails to send the notifications back onto a queue with a logged message to be retried later, and after a few tries, it just gets abandoned?  Do you think we should track success counts vs failures? Right now in terraform I don't have these logs forwarded to splunk, but we could.

Reply:
I was thinking back and forth on this quite a bit, and I've decided I think it's actually fine as is, but it would be great if we could get something set up in cloudwatch or something to track what status codes get logged in the error output (from "Failed to send message. Status code: {response.status_code}"). So that way we can have metrics available to see if/when someone brings an issue with subscriptions to us we can check the chart and see if there was a sudden surge in http error codes.
My concern is that if we just let the lambda succeed and don't have any note anywhere about what error got logged, we'll have a much harder time investigating any major issues someone might bring to us

Reply:
I don't think exit() is the best way, just doing a google and an AI request

Reply:
No, I think letting the lambda just print to the error output and have the lambda itself succeed is fine, but think it would be great if there's some way we could capture the metrics of how many error codes we get (though if the team doesn't think we need that, that's also fine)